src/util/expr.h

```diff
@@ -319,6 +319,29 @@ class exprt:public irept
   const_unique_depth_iteratort unique_depth_cend() const;
 };
 
+/// Class for expressions that offers a restriction of the API
+/// offered by exprt. Notably, access to non-type-safe methods such
+/// as opX() is protected.
```

Comment:
Does that mean that `exprt` subclasses that so far _only_ have these as their API will get updated (e.g. classes like `typecast_exprt` that only have `op0` rather than a named accessor?).

Also, if we're already going to do an API do-over, why make the "new" expr inherit from `irept` at all, rather than have an irep as a protected field or something?